hardware/raspberrypi/bootmodes/bootflow.md

```diff
@@ -2,19 +2,13 @@
 
 **The following boot sequence applies to the BCM2837-based models of Raspberry Pi only. On all other models, the Pi will try [SD card boot](sdcard.md), followed by [USB device mode boot](device.md).**
 
-The boot sequence begins with reading the OTP to determine which boot modes are enabled. By default, this is SD card boot followed by USB device boot. Subsequently, the boot ROM checks to see whether the GPIO boot mode OTP bits have been programmed — one to enable GPIO boot mode, and one to select the bank of GPIOs it uses to disable boot modes (low = GPIOs 22-26, high = GPIOs 39-43). This makes it possible to use hardware attached to the GPIO connector to choose between different boot modes.
-
-The GPIO boot mode OTP bits can be programmed by adding `program_gpio_bootmode=n` to `config.txt`, where n is `1` to select the low bank (22-26) or `2` to select the high bank (39-43). Once added, boot the device, then power-cycle it (rebooting is not sufficient). You should expect it to no longer boot (all boot modes will be disabled by default). Apply a pull-up to the required pin to enable the required boot mode. After programming, the `config.txt` setting can be removed.
-
-N.B.:
-1. It is important to remember that, **once set, OTP bits can never be unset**, so think carefully before enabling this facility because it effectively makes 5 GPIOs unusable for other purposes. Also note that the bit assignments make it possible to switch from the low bank (22-26) to the high bank (39-43), **but not back again**, and that selecting the high bank is likely to produce a non-bootable Pi, unless you're using a Compute Module (any version).
-2. **Do not use `program_gpio_bootmode` unless your firmware is dated 20 Oct 2017 or later (`vcgencmd version`). Doing so will make it impossible to select the low bank of boot mode GPIOs.**
+When the BCM2837 boots, it uses two different sources to determine which boot modes to enable - firstly, the OTP is checked to see which boot modes are enabled. It then checks to see if GPIO boot mode is enabled. If GPIO boot mode is enabled, then the relevant GPIO lines are tested to select which of the OTP-enabled boot modes should be attempted. If GPIO boot mode is disabled, then the boot modes selected in the OTP are final, and no probing of the GPIO lines is necessary. Note that GPIO boot mode can only be used to select boot modes which are already enabled in the OTP. See [GPIO boot mode](gpio.md) for details on configuring GPIO boot mode. GPIO boot mode is disabled by default.
```

Comment:
"firstly, the OTP is checked to see which boot modes are enabled. It then checks to see if GPIO boot mode is enabled."

Isn't GPIO boot mode also an OTP setting? Perhaps this paragraph could be tidied up a bit?

Reply:
Yes, GPIO boot mode is indeed also an OTP setting. But since it is also used as a second stage to determine which boot modes to enable, I mention it in that order. That paragraph is indeed a mess, but at the time that was me best effort at getting to to say what it needs to say. Maybe a list of steps like the main bootflow page would be better?

Reply:
What if you took out the "It then checks to see if GPIO boot mode is enabled." sentence entirely (it feels a bit clunky) and changed the next sentence to start "If the GPIO boot mode setting is enabled, " ?
Perhaps the "If GPIO boot mode is disabled, then the boot modes selected in the OTP are final, and no probing of the GPIO lines is necessary." sentence is also unnecessary?